### ISSUE TITLE
switching from lexer to parser in a split grammar regenerates `.tokens` no matter what

### ISSUE THREAD

**Issue by parrt** [antlr/intellij-plugin-v4#540]
This could slow down switching between tabs so we should check see if the .tokens file is older than the lexer .g4 file in `RunANTLROnGrammarFile.run() on line 85.

```java
	// is lexer file? gen .tokens file no matter what as tokens might have changed;
	// a parser that feeds off of that file will need to see the changes.
	if ( previewState.g==null && previewState.lg!=null) {
		Grammar g = previewState.lg;
		String language = g.getOptionString(ANTLRv4GrammarProperties.PROP_LANGUAGE);
		Tool tool = ParsingUtils.createANTLRToolForLoadingGrammars(getGrammarProperties(project, grammarFile));
		CodeGenerator gen = CodeGenerator.create(tool, g, language);
		gen.writeVocabFile();
	}
```

**Comment by kjarrio** [comment]
I have a fix for this one, but still figuring out how to properly test it in a headless environment

**Comment by kjarrio** [comment]
Should be fixed in #577

**Comment by parrt** [comment]
Fixed by https://github.com/antlr/intellij-plugin-v4/pull/577
